### PR TITLE
[refactor] Remove `test_quantizer.py` and reuse one of its test in `test_quantization.py`

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#27269 [refactor] Remove `test_quantizer.py` and reuse one of its test in `test_quantization.py`**
* #27239 [refactor] Factor out insertPrepackUnpackForLinear
* #27238 [refactor] InsertObservers
* #27237 [quant][graphmode][fix] Make inserted child module names unique

Summary:
Remove `test_quantizer.py`, add and rewrite one of the tests in `test_quantizer`
in `test_quantization.py`
The conv test is removed for now since conv pattern is still broken, we'll add another test
later

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18182916](https://our.internmc.facebook.com/intern/diff/D18182916)

### CODE REVIEW

**Comment by ZolotukhinM** [comment]
The added test looks differently from the two tests that were removed. Could you please explain in more details why this is equivalent (also, please add that to the commit message since right now it doesn't look like the test was moved)?

**Comment by jerryzh168** [comment]
for test_quantizer.py, the first test is removed, the second one is the one we move to test_quantization, and I also modified a bit to make it pass

File: test/test_quantization.py [hunk 1]
```diff
@@ -695,10 +696,49 @@ def test_single_layer(self):
             [self.calib_data],
             inplace=False)
         result_eager = model_eager(self.calib_data[0][0])
-        torch._C._jit_pass_quant_fusion(model_script._c._get_module('fc1')._get_method('forward').graph)
-        result_script = model_script._c._get_method('forward')(self.calib_data[0][0])
+        result_script = get_forward(model_script._c)(self.calib_data[0][0])
```
Comment (ZolotukhinM) [inline test/test_quantization.py]: Why did this change if that was just a move of the test?

File: test/test_quantization.py [hunk 2]
```diff
@@ -695,10 +696,49 @@ def test_single_layer(self):
...
         self.assertEqual(result_eager, result_script)
 
+    @unittest.skip("quantization for inlined linear is not working right now")
+    def test_nested(self):
```
Comment (ZolotukhinM) [inline test/test_quantization.py]: It seems that we had more tests before the move, could you please explain why they were changed?

**Review by ZolotukhinM — CHANGES_REQUESTED** [review]

File: test/test_quantization.py (hunk 1 above)
Comment (jerryzh168) [reply test/test_quantization.py]: this is move + some modifications to make it pass I'll modify the message

File: test/test_quantization.py (hunk 2 above)
Comment (jerryzh168) [reply test/test_quantization.py]: the conv+bn test is added later after we fix the conv patterns

**Review by ZolotukhinM — APPROVED** [review]
Looks good! In future when you move stuff around please try to split it from non-mechanical changes for easier reviews.

**Pull request closed without merging**